vendor/github.com/docker/libnetwork/agent.go

```diff
@@ -463,52 +463,64 @@ func (ep *endpoint) deleteDriverInfoFromCluster() error {
 }
 
 func (ep *endpoint) addServiceInfoToCluster() error {
+	if ep.isAnonymous() && len(ep.myAliases) == 0 || ep.Iface().Address() == nil {
```

Comment:
If you are referring to the case where `ep.myAliases==nil`, it would be fine because `len(ep.myAliases)`  would be `0`. The check is in fact for covering both the case of an empty and an uninitialized list of aliases.